### PR TITLE
Fb sync ww3

### PR DESCRIPTION
**Pull request by aliabdolali**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description
* sync WW3 subcomponent to the top of WW3 repo.
* mod_def files are created and added to Hera and Orion. 
* the tests with wave components are changed due to changes in mod_defs. 

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

The tests with wave components ran successfully on Orian with Intel compiler. I am waiting until knowing the commit date, so I can run the entire tests and creat baselines for the tests with wave.

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
Please fill out the entire template, without deleting sections. When filling out the PR checklist note the results of any testing that has been done. Thanks.

**Comment by JessicaMeixner-NOAA** [comment]
I have tested this PR with hera.intel and all regtests passed.  I'll continue testing on other platforms and report back.

**Comment by JessicaMeixner-NOAA** [comment]
I have had successful tests with: 
hera.intel 
wcoss-dell 
orion.intel 

My test on gaea timed out so I'm re-running those but do not anticipate any issues.   Note that the "input-data-20210825/WW3_input_data_20210908" needs to be copied to new machines (it was already on orion, I copied it to gaea and to wcoss-dell).

**Comment by DeniseWorthen** [comment]
@aliabdolali We would like to add the memory profiling for WW3 as described in this [discussion](https://github.com/ufs-community/ufs-weather-model/discussions/779). Would you also be able to include the memory profile [branch](https://github.com/DeniseWorthen/WW3/tree/feature/memprofile2) I created in this PR? 

Also, is there a reason that the corner bug fix PR from Ufuk is not being included with this update? This update points to WW3  a009776 which is prior to the corner bug fix merge, correct?

**Comment by aliabdolali** [comment]
Hi @DeniseWorthen 
The FV3 and ufs-weather-model are merged in this PR, and the corner bugfix PR from Ufuk is now in this PR (It was not yet in WW3 main repo when I initiated this PR). 
Yes, we can add your PR for memory profiling to this PR.

**Comment by JessicaMeixner-NOAA** [comment]
A PR with the updates from @DeniseWorthen for the ESMF memory profiling: https://github.com/NOAA-EMC/WW3/pull/473

File: tests/rt.sh [hunk 1]
```diff
@@ -415,15 +415,15 @@ if [[ $TESTS_FILE =~ '35d' ]]; then
   TEST_35D=true
 fi
 
-BL_DATE=20210915
+BL_DATE=20210921
```
Comment (JessicaMeixner-NOAA) [inline tests/rt.sh]: Do answers change for this PR? I thought the input was updated but that the answers were the same?

Comment (aliabdolali) [reply tests/rt.sh]: I am going to check, will revert it if not applicable.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/730785387/20210921150018/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_15113
Test datm_cdeps_mx025_cfsr 089 failed failed 
Test datm_cdeps_mx025_cfsr 089 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-BL

**Comment by aliabdolali** [comment]
> _**Automated RT Failure Notification**_
> Machine: hera
> Compiler: intel
> Job: BL
> Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/730785387/20210921150018/ufs-weather-model
> Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_15113
> Test datm_cdeps_mx025_cfsr 089 failed failed
> Test datm_cdeps_mx025_cfsr 089 failed in run_test failed
> Please make changes and add the following label back:
> hera-intel-BL

Do I need to do it? 
I have no permission to do it.

**Comment by DusanJovic-NOAA** [comment]
> _**Automated RT Failure Notification**_
> Machine: hera
> Compiler: intel
> Job: BL
> Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/730785387/20210921150018/ufs-weather-model
> Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_15113
> Test datm_cdeps_mx025_cfsr 089 failed failed
> Test datm_cdeps_mx025_cfsr 089 failed in run_test failed
> Please make changes and add the following label back:
> hera-intel-BL

```
31 min. TEST 089 datm_cdeps_mx025_cfsr is running,  status: R jobid 23392753
32 min. TEST 089 datm_cdeps_mx025_cfsr is running,  status: R jobid 23392753
Slurm unknown status -. Check sacct ...
23392753                  TIMEOUT         rt_15113_089 
23392753.ba+            CANCELLED                batch 
23392753.ex+            COMPLETED               extern 
23392753.0                 FAILED              fv3.exe 
33 min. TEST 089 datm_cdeps_mx025_cfsr is TIMEOUT,  status: - jobid 23392753

```

File: tests/rt.sh (hunk 1 above)
Comment (DusanJovic-NOAA) [reply tests/rt.sh]: Do we need new baselines or not?

Comment (aliabdolali) [reply tests/rt.sh]: The tests ran successfully with -c option. Now, I submitted the wave tests to be compared with the baseline, but my job is still in the queue.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/730785387/20210921094508/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_152435
Please make changes and add the following label back:
orion-intel-BL

**Comment by DusanJovic-NOAA** [comment]
> _**Automated RT Failure Notification**_
> Machine: orion
> Compiler: intel
> Job: BL
> Repo location: /work/noaa/nems/emc.nemspara/autort/pr/730785387/20210921094508/ufs-weather-model
> Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_152435
> Please make changes and add the following label back:
> orion-intel-BL

I do not understand what's wrong here:
```
INFO:BL/PROCESS_LOGFILE:It is: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_152435
INFO:BL/PROCESS_LOGFILE:RT Successful
INFO:BL/CHECK_FOR_BL_DIR:Checking if baseline directory exists
CRITICAL:JOB/RUN:run() FAILED. Exception:<class 'Exception'>
INFO:JOB/RUN:Sending comment text
INFO:JOB/SEND_COMMENT_TEXT:Comment Text: ***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/730785387/20210921094508/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_152435

INFO:MAIN:Script Finished
```
the only thing I see is that run() failed due to `Exception:<class 'Exception'>`.

**Comment by BrianCurtis-NOAA** [comment]
That's definitely bad coding on my part. I'm guessing it checked for the baseline directory and found it to exist and stopped (in case it were to overwrite good baseline data)

**Comment by DusanJovic-NOAA** [comment]
> That's definitely bad coding on my part. I'm guessing it checked for the baseline directory and found it to exist and stopped (in case it were to overwrite good baseline data)

How is this possible, the baseline date is new, the directory should not exist.

```
drwxrwsr-x 37 junwang  nems  4096 Sep  9 11:00 input-data-20210825
drwxrwsr-x  3 dworthen nems  4096 Sep 17 04:07 develop-20210915
drwxr-s---  3 bcurtis  nems  4096 Sep 21 12:09 develop-20210921
```
Directory `develop-20210921` has been created at 12:09 today. And it's empty.

**Comment by junwang-noaa** [comment]
It looks to me the baseline was created successfully, but it can't be
copied to
/work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20210921/INTEL. Not sure
if it's a permission issue. We need to manually copy the baseline to the
directory.

File: tests/rt.sh (hunk 1 above)
Comment (aliabdolali) [reply tests/rt.sh]: @DusanJovic-NOAA it seems I was wrong and the baseline does not need to be changed. Do you want me to revert it to 20210915?

Comment (DusanJovic-NOAA) [reply tests/rt.sh]: Yes. Revert BL_DATE in rt.sh.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/730785387/20210921144505/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_35127
Baseline creation and move successful
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/730785387/20210921181019/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_3276
Test regional_quilt 037 failed in run_test failed 
Test hafs_regional_atm 084 failed in run_test failed 
Test cpld_bmark_v16_nsst 012 failed in run_test failed 
Test control_c384gdas_wav 107 failed in run_test failed 
Test datm_cdeps_restart_cfsr 098 failed in run_test failed 
Test hafs_regional_datm_cdeps 088 failed in run_test failed 
Test cpld_bmark_v16 010 failed in run_test failed 
Test cpld_control_c384 008 failed in run_test failed 
Test hafs_regional_docn_oisst 087 failed in run_test failed 
Test cpld_2threads 003 failed in run_test failed 
Test hafs_regional_docn 086 failed in run_test failed 
Test regional_restart 036 failed in run_test failed 
Test datm_debug_cfsr 096 failed in run_test failed 
Test cpld_decomp 004 failed in run_test failed 
Test cpld_ca 005 failed in run_test failed 
Test control_atmwav 106 failed in run_test failed 
Test cpld_control_c192 006 failed in run_test failed 
Test cpld_control 001 failed in run_test failed 
Test datm_restart_cfsr 090 failed in run_test failed 
Test control_flake 049 failed in run_test failed 
Test cpld_bmark_wave_v16_p7b 014 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-BL

**Comment by aliabdolali** [comment]
Hi @DusanJovic-NOAA
is the job on Cray done? My job on dell is still running (since yesterday). There are 5 tests that remained but waiting in the queue.

**Comment by DusanJovic-NOAA** [comment]
> Hi @DusanJovic-NOAA
> is the job on Cray done? My job on dell is still running (since yesterday). There are 5 tests that remained but waiting in the queue.

Yes, it is. Please commit these two files:

/gpfs/dell2/emc/modeling/noscrub/Dusan.Jovic/ufs/pr790/ufs-weather-model/tests/RegressionTests_wcoss_dell_p3.log

and

/gpfs/hps3/emc/meso/save/Dusan.Jovic/ufs/pr790/ufs-weather-model/tests/RegressionTests_wcoss_cray.log

**Comment by aliabdolali** [comment]
> > Hi @DusanJovic-NOAA
> > is the job on Cray done? My job on dell is still running (since yesterday). There are 5 tests that remained but waiting in the queue.
> 
> Yes, it is. Please commit these two files:
> 
> /gpfs/dell2/emc/modeling/noscrub/Dusan.Jovic/ufs/pr790/ufs-weather-model/tests/RegressionTests_wcoss_dell_p3.log
> 
> and
> 
> /gpfs/hps3/emc/meso/save/Dusan.Jovic/ufs/pr790/ufs-weather-model/tests/RegressionTests_wcoss_cray.log

Done

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Pull request merged**